main/src/library/Eff/Random.flix

```diff
@@ -0,0 +1,54 @@
+/*
+ *  Copyright 2024 Magnus Madsen
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+mod Eff.Random {
+
+    pub eff RandomCoin {
+        ///
+        /// Returns `true` or `false` with equal chance.
```

Comment:
I thought this was up to the handler?

Reply:
That's a complex philosophical question. 

If it is purely up to the handler, then what can you even say about an effect? That is has no meaning other than its type?

No, I think the point is that under IO semantics, the meaning of RandomCoin.flip is to return a uniformly distributed Boolean. Similarly, the meaning of BiasedCoin.flip is to return a biased Boolean.

A handler may change or enrich that meaning, but it must do so in a way that is meaningful w.r.t. the original program.

Reply:
A handler may __re-interpret__ an effect, but that does not mean the original effect does not have a meaning by itself.

Reply:
Philosophically, I would not mix up probability (suggested by the name "Random") with non-determinism.

Historically, the specification would be in terms of algebraic equations. That is, you would have, for example, laws on how `flip` interacts with `raise`.

- `if (flip) raise else e` = `e`
- `if (flip) e else raise` = `e`